### DISCUSSION TITLE
Why is there no way to run all tests in a class as separate fixtures (and thus in parallel)?

### DISCUSSION THREAD

**Discussion by worthy7** [xunit/xunit#2474] [Question] [answered]
Tests are supposed to be unique and 100% isolated from each other. Ok so I've done that.
My setup code for the "context" is in the class constructor.

But there are no options for allowing xunit to run all these tests in parallel. Why not?

```
/// <summary>
    /// Proof that tests are run isolated at least in the c# environment
    /// </summary>
    public class XunitTests
    {
        int runcount = 0;
        int setuptimes = 0;
        public XunitTests()
        {
            setuptimes++;
        }
        [Fact]
        public async Task Something1()
        {
            runcount++;
            Thread.Sleep(5000);
            runcount.ShouldBe(1);
            setuptimes.ShouldBe(1);
        }
        [Fact]
        public async Task Something2()
        {

            runcount++;
            Thread.Sleep(3000);
            runcount.ShouldBe(1);
            setuptimes.ShouldBe(1);
        }
    }
```

Why isn't this run in parallel by default? Why is the default to run them sequentially, even though they are not sharing any context? It doesn't make sense to me. Why can't his take 5 seconds instead of 8? And why isn't there at least an option to allow that (since I think it should be the default).

**Comment by bradwilson** [comment, marked as answer]
#1986 
#2055 

One or both of those should cover your request.
